nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -979,20 +980,28 @@ def get_user_full_name(self, userid):
         """
             Get the last contributor full name
         """
-        fullname = userid
         try:
-            if userid in self._user_cache:
-                fullname = self._user_cache[userid]
-            else:
-                rest_client = self.get_rest_api_client()
-                response = rest_client.get_user_full_name(userid)
-                if response and 'properties' in response:
-                    properties = response['properties']
-                    firstName = properties.get('firstName')
-                    lastName = properties.get('lastName')
-                    if firstName and lastName:
-                        fullname = " ".join([firstName, lastName]).strip()
-                        self._user_cache[userid] = fullname
+            rest_client = self.get_rest_api_client()
+            response = rest_client.get_user_full_name(userid)
+            if response and 'properties' in response:
+                properties = response['properties']
+                firstName = properties.get('firstName')
+                lastName = properties.get('lastName')
+                self._dao.insert_update_user_info(userid, firstName, lastName)
         except urllib2.URLError as e:
             log.exception(e)
-        return fullname
+        return self._dao.get_user_info(userid)
+
```

Comment:
The fetch_all_users method is supposed to fetch all userid  to full name mapping for the tenant. The API used to do this has an upper limit of 50 users, so if userid is not within the first 50 users, we have to individually resolve the userid.